### DISCUSSION TITLE
Esp stop working after 10 minutes

### DISCUSSION THREAD

**Discussion by Thebasestar** [gekkekoe/esphome-ecodan-hp#62] [General]
Hi as for title from 3 days my esp stop sending data after 10 minutes from boot, and it became unreachable from wifi, after a manual reset or disconnection it restart correctly but hang again after 10 min.... can it be faulty?, or is something changed in the setup of yaml?I flashed last version, but i'm still using wifi setting in the main yaml and not using separate wifi.yaml

Thanks,
Base

**Comment by gekkekoe** [comment]
Best is to download latest ecodan-esphome.yaml and apply your custom settings. The wifi part has been separated (in order to support ethernet). So it's best to use the latest yamls.

But if it did work for 10 mins, then it's probably bad wifi signal. Esphome will go into safe mode after x times trying to reconnect due to bad signal/noise. Almost all connectivity reports I have seen, had this. Probably it tries to connect to an access point that's not close. You can verify if this is the case. Some AP support client pinning, then you can bind the client to a certain access point. 

For the first 10 mins, what is your wifi signal db?

If you cannot fix the wifi (via pinning or installing an access point near the heatpump), you could also try to use ethernet. Get one of the esp32 boards with integrated ethernet and check if they are supported by esphome. Then just use ethernet (look at heishamon how it's configured).

going to convert this to a discussion, this is not an issue.

**Reply by Thebasestar** [reply]
Many thanks for the advices and suggestions, solved rebooting the AP on the basement floor....
